checkstyle/suppressions.xml

```diff
@@ -56,6 +56,11 @@
     <suppress checks="MethodCount" files="com/hazelcast/cache/impl/CacheStatisticsImpl"/>
     <suppress checks="ClassDataAbstractionCoupling" files="com/hazelcast/cache/impl/DefaultOperationProvider"/>
 
+    <!-- Cardinality -->
+    <suppress checks="JavadocMethod" files="com/hazelcast/cardinality/"/>
```

Comment:
I agree, some minor suppressions for internal classes should be allowed to be added package-wide but I would say that other code smell warnings for both private and public classes should be defined with annotations.